### PR TITLE
Fix transposed matrix

### PR DESCRIPTION
**Pull request by eriknw**
Don't subclass TransposedMatrix from Matrix.  It's not really a GbContainer.  This led to surprising things to be possible.

Instead of blacklisting methods that don't work, choose to be positive and add the ones that do.  I hope this will make it harder to make mistakes in the future.

I took the liberty of moving `is_scalar` and `is_transposed` to private properties.

This PR continues from: https://github.com/jim22k/grblas/pull/15

### CODE REVIEW

File: grblas/base.py [hunk 1]
```diff
@@ -49,20 +49,16 @@ def update(self, delayed):
...
         if not isinstance(gb_obj, ffi.CData):
             raise TypeError('Object passed to __init__ must be CData type')
         if not isinstance(dtype, dtypes.DataType):
             dtype = dtypes.lookup(dtype)
-
         self.gb_obj = gb_obj
         self.dtype = dtype
 
-    def __invert__(self):
-        return ComplementedMask(self)
```
Comment (jim22k) [inline grblas/base.py:64]: Removing this means a user cannot `(~M).V`. Probably not worth allowing.
If we are removing this, we should remove the ComplementedMask class.

Should we give a useful error message if someone tries to `mask=~M`? A message that masks need to indicate either `.V` or `.S`?

File: grblas/matrix.py [hunk 2]
```diff
@@ -561,62 +556,67 @@ def _prep_for_assign(self, resolved_indexes, obj):
                     raise TypeError(f'Expected Matrix for assignment value; found {type(obj)}')
                 delayed = GbDelayed(lib.GrB_Matrix_assign,
                                     [obj.gb_obj[0], rows, rowsize, cols, colsize],
-                                    at=obj.is_transposed)
-
+                                    at=obj._is_transposed)
```
Comment (jim22k) [inline grblas/matrix.py:559]: `isinstance(obj, Matrix)` check above needs to also check for TransposedMatrix

Comment (eriknw) [reply grblas/matrix.py:559]: Good catch!  Fixed and added a test to exercise this.

File: grblas/base.py (hunk 1 above)
Comment (eriknw) [reply grblas/base.py:64]: Good call on `ComplementedMask` (a coverage report would have hinted at this).  A useful error message here may not hurt.

**Pull request merged**